plugins/outputs/clarify/clarify.go

```diff
@@ -0,0 +1,197 @@
+		if err != nil {
+			return err
+		}
+		c.client = creds.Client(ctx)
+		return nil
+	case !c.Username.Empty() && !c.Password.Empty():
+		username, err := c.Username.Get()
+		if err != nil {
+			return fmt.Errorf("getting username failed: %w", err)
+		}
+		password, err := c.Password.Get()
+		if err != nil {
+			config.ReleaseSecret(username)
+			return fmt.Errorf("getting password failed: %w", err)
+		}
+		creds := clarify.BasicAuthCredentials(string(username), string(password))
+		c.client = creds.Client(ctx)
+		config.ReleaseSecret(username)
+		config.ReleaseSecret(password)
+		return nil
+	}
+	return fmt.Errorf("no Clarify credentials provided")
+}
+
+func (c *Clarify) Connect() error {
+	return nil
+}
+
+func toFloat64(v interface{}) (float64, error) {
+	var value float64
+	switch v := v.(type) {
+	case bool:
+		value = float64(0)
+		if v {
+			value = float64(1)
+		}
+		return value, nil
+	default:
+		return internal.ToFloat64(v)
+	}
+}
+
+func (c *Clarify) Write(metrics []telegraf.Metric) error {
+	frame, signals := c.processMetrics(metrics)
+
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, time.Duration(c.Timeout))
+	defer cancel()
+
+	if _, err := c.client.Insert(frame).Do(ctx); err != nil {
+		return err
+	}
+
+	if _, err := c.client.SaveSignals(signals).Do(ctx); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Clarify) processMetrics(metrics []telegraf.Metric) (views.DataFrame, map[string]views.SignalSave) {
+	signals := make(map[string]views.SignalSave)
+	frame := views.DataFrame{}
+
+	for _, m := range metrics {
+		for _, f := range m.FieldList() {
+			value, err := toFloat64(f.Value)
+			if err != nil {
+				c.Log.Warnf("Unable to add field `%s` for metric `%s` due to error '%v', skipping", f.Key, m.Name(), err)
+				continue
+			}
+			id, err := c.generateID(m, f)
+			if err != nil {
+				c.Log.Warnf("Unable to add field `%s` for metric `%s` due to error '%v', skipping", f.Key, m.Name(), err)
+				continue
+			}
+			ts := fields.AsTimestamp(m.Time())
+
+			if _, ok := frame[id]; ok {
+				frame[id][ts] = value
+			} else {
+				frame[id] = views.DataSeries{ts: value}
+			}
+
+			s := views.SignalSave{}
+			s.Name = m.Name() + "." + f.Key
+
+			for _, t := range m.TagList() {
+				labelName := strings.ReplaceAll(t.Key, " ", "-")
+				labelName = strings.ReplaceAll(labelName, "_", "-")
+				labelName = strings.ToLower(labelName)
+				s.Labels.Add(labelName, t.Value)
+			}
+
+			signals[id] = s
+		}
+	}
+	return frame, signals
+}
+
+func normalizeID(id string) string {
+	return strings.Map(func(r rune) rune {
+		if strings.ContainsRune(allowedIDRunes, r) {
+			return r
+		}
+		return '_'
+	}, id)
+}
+
+func (c *Clarify) generateID(m telegraf.Metric, f *telegraf.Field) (string, error) {
+	var id, cid string
+	exist := false
+	if c.ClarifyIDTag != "" {
+		cid, exist = m.GetTag(c.ClarifyIDTag)
+	}
+	if exist && len(m.FieldList()) == 1 {
+		id = cid
+	} else {
```

Comment:
Yes. This is because the plugin is used for a lot of metrics with one field where the ID is defined in a specific tag. Having this option here allows for the ID to come from the source system.

Reply:
I agree that this could be solved by using a processor, but I feel it's both easier and more clean to keep it like this. And it doesn't add much overhead to either config or code. 

A good example where one needs to use `id_tags` array is when using the `[[inputs.temp]]` input plugin, where the metric name is temp, field name is value and the only thing distinguishing the signals would be the `sensor` tag.

Reply:
To me that looks unexpected and also undocumented behaviour..

If one sets a clarify ID tag, but has a metric with multiple fields (as most do) then this setting will be ignored silently..

Reply:
It's documented in the README, but agree that it can be a bit more clear.
Suggested addition:
---
If you wish to specify a specific tag to use as the input id, set the config
option `clarify_id_tag` to the tag containing the id to be used.
If  `clarify_id_tag = 'clarify_input_id'` and this tag is present and there
is only one field present in the metric, this tag will be used as the inputID
in Clarify. **If there is more fields available in the metric the `clarify_id_tag`
will be ignored and normal id generation will be used.**

Reply:
Yeah, that will make it more clear. I would however remove this exact part:
> `clarify_id_tag = 'clarify_input_id'` and

So it becomes:
> If this tag is present and there is only one field present in the metric, this tag will be used as the inputID in Clarify. If there are more fields available in the metric, the tag will be ignored and normal id generation will be used.